BedrockServer.cpp

```diff
@@ -213,14 +205,7 @@ void BedrockServer::sync(const SData& args,
     SINFO("Starting " << workerThreads << " worker threads.");
     list<thread> workerThreadList;
     for (int threadId = 0; threadId < workerThreads; threadId++) {
-        workerThreadList.emplace_back(worker,
-                                      ref(dbPool),
-                                      ref(replicationState),
-                                      ref(leaderVersion),
-                                      ref(syncNodeQueuedCommands),
-                                      ref(server._completedCommands),
-                                      ref(server),
-                                      threadId);
+        workerThreadList.emplace_back(&BedrockServer::worker, this, threadId);
```

Comment:
so looking at what emplace_back does https://www.cplusplus.com/reference/list/list/emplace_back/ it should work like `push_back` to place the worker at the end of the `workerThreadList` and since you're passing a class it's construct a new `BedrockServer::worker` with `this` and `threadID`, right?

I'm a little confused why `this` is needed, probably something to do with passing the `BedrockServer::worker` by reference? Or b/c `BedrockServer::worker` is a method?

Also should it be `this` or `*this`, I noticed other usages were `*this` like here https://github.com/Expensify/Bedrock/blob/6d7c1af6b46c33be76c97564091893619d1a840e/BedrockServer.cpp#L741

Reply:
Oh, and it takes `this` instead of `*this` on purpose. The first is the `this pointer`, which is exactly the same thing that this implicit parameter to member functions always is. The second dereferences the `this pointer` to pass a reference to the current object.

If you were to look at the assembly output, passing `this` by value or `*this` by reference probably end up looking identical.